### PR TITLE
Questions page

### PR DESCRIPTION
**Pull request by foolhardy21**
## Rules and Questions Page.

This PR has the rules page and questions page. The rules and questions page shown are based on the category selected from the home page.

- Files to review:
  - [ ] src/pages/Rules.jsx
  - [ ] src/pages/Questions.jsx
  - [ ] src/data/questions.js
  - [ ] src/data/rules.js
 
- Use this link for preview -  https://deploy-preview-5--stupefied-hawking-9910e7.netlify.app/

Screenshot (37)
v-1
Thanks

### CODE REVIEW

File: src/pages/Rules.jsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    useEffect(() => {
+        const category = params.category
+        const categoryRules = rules.find(rulesSection => rulesSection.category === category)
+        setRulesState(categoryRules)
+    }, [])
+
+    return (
+        <div
+            style={{
```
Comment (mikesch803) [inline src/pages/Rules.jsx:21]: suggestion: you can avoid using inline-style, instead you can add to className

File: src/pages/Questions.jsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                        <p>{score}/{categoryQuestions.length}</p>
+                        <p className={`txt-md card-txtw-md ${getTextColor(theme)} txt-cap mg-btm-xs`}>
+                            {currentQues?.question}
+                        </p>
+                        {
+                            params.category === 'guess' && <div className="flx flx-center mg-xs"><img srcSet={currentQues.img} alt="Sneaker" className={styles.quesImg} /></div>
+                        }
+                        {
+                            currentQues?.options?.map((option, index) =>
+                                <button key={index} ref={optionBtnRefs.current[index]} onClick={() => handleOptionSelect(optionBtnRefs.current[index])} value={option} className="btn-outlined txt-md txt-cap b-solid b-primary pd-xs mg-btm-xs">{option}</button>)
```
Comment (mikesch803) [inline src/pages/Questions.jsx:74]: suggestion : this could be done using state

**Review by mikesch803 — APPROVED** [review]
I put some suggestion, else everything is good

File: src/pages/Questions.jsx (hunk 2 above)
Comment (foolhardy21) [reply src/pages/Questions.jsx:74]: As I am changing the style of the selected option, I have used `useRef`. Can you tell a bit more about what you are suggesting ?

File: src/data/categories.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import hS from 'assets/h-s.jpg'
+import nbaM from 'assets/nba-m.jpg'
+import nbaS from 'assets/nba-s.jpg'
+import sneakM from 'assets/sneak-m.jpg'
+import sneakS from 'assets/sneak-s.jpg'
+
+export const categories = [
+    {
+        name: 'hiphop',
+        srcSet: `${hM} 400w, ${hS} 250w`
```
Comment (VLeads) [inline src/data/categories.js:11]: appreciate, you are using responsive images !!

**Review by VLeads — APPROVED** [review]
Not much to add. All good !!

**Pull request merged**